File: leetcode/1108_defanging-an-ip-address/1.py
# https://leetcode.com/problems/defanging-an-ip-address/

# trial 1
from typing import List, Dict, Tuple, Sequence
def determine(x):
    if x == '.': return "[.]"
    else: return x              
class Solution:
    def defangIPaddr(self, address: str) -> str:
        defanged = ''
        for s in address:
            defanged = defanged + determine(s)
        return defanged
t = Solution()
t.defangIPaddr("255.100.50.0")

File: ltc/1108_defanging-an-ip-address/1.clj
; https://leetcode.com/problems/defanging-an-ip-address/

(defn defanging-an-ip-address [address]
  (clojure.string/join 
   (map
    (fn [c] (if (= c \.) "[.]" c))
    (char-array address))))

(defanging-an-ip-address "1.1.1.1") ; returns "1[.]1[.]1[.]1"
(defanging-an-ip-address "255.100.50.0") ; returns "255[.]100[.]50[.]0"